scripts/wod.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+//pool, explode-at, rote
+//where rote means the intial failed dice are rerolled
+//notably, those rerolled dice cannot explode, and failed exploding dice from qualities can't either
+//meaning that rote has to be counted separately AFTER accounting for dice explosion
+//All 10s explode
+//Okay new idea, rote just adds some more dice that are checked for success at the report, 
+//but not checked by explode or anything. 
+
+const SUCCESS_THRESHOLD = 8;
+const DEFAULT_EXPLODE_TARGET = 10;
+
+module.exports = function(robot) {
+	robot.respond(/roll (\d+)(?:e(\d+))?(r?)/i, function (msg) {
+		let answer;
+		let pool = parseInt(msg.match[1]);
+  	let explode_target = parseInt(msg.match[2]) || DEFAULT_EXPLODE_TARGET;
+  	let is_rote = Boolean(msg.match[3]);
+  	if (pool < 1) {
+  		answer = 'That is too few dice, my human';
+  	} else if (pool > 100){
+  		answer = 'That is too many dice, my human';
+  	} else if (explode_target < 8) {
+  		answer = 'That is an invalid explode target, human';
+  	} else if (is_rote) {
+  		let initial_roll = rote(rollMany(pool));
+  		let explode_allowed = (explode(rollMany(pool), explode_target));
+  		Array.prototype.push.apply(explode_allowed, initial_roll);
+  		answer = report(explode_allowed);
+  	} else {
+  		answer = report(explode(rollMany(pool), explode_target));
+  	}
+  	msg.reply(answer);
+	});
+}
+
+
+function roll() {
+	let result = Math.floor(Math.random()*10) + 1;
+	return result ;
+}
+
+
+function rollMany(dice){
+	let results = [];
+	for (var i = 0; i < dice; i++) {
+		results.push(roll());
+	}
+	results.sort(function(a, b) {
+		return a - b;
+	});
+	return results;
+}
+
+
+function explode(results, target) {
+	let myArray = results.slice(0);
+	for (var num of myArray) {
+		if (num >= target){
+			myArray.push(roll());
+		}
+	}
+	return myArray;
+}
+
+
+function rote(results) {
+	let reroll = results.slice(0);
+	let noExplodeyDice = [];
+	for (var i = 0; i < reroll.length; i++) {
+		if (reroll[i] < 8){
+			noExplodeyDice.push(roll()); //do we want push here, or assignment at [i]?
+		}
+	}
+	return noExplodeyDice;
```

Comment:
So, I tried rote out and got these strange results:

```
tyche> tyche roll 1e8r
Shell: I rolled a 3.
tyche> tyche roll 1e8r
Shell: I rolled 7 and 1 making 0 hits.
```

I think that the change in the call to rote that I mention above should resolve this.